relayer/src/supervisor/client_state_filter.rs

```diff
@@ -0,0 +1,195 @@
+use std::collections::HashMap;
+use tracing::debug;
+
+use crate::object;
+use crate::registry::Registry;
+use anomaly::BoxError;
+use ibc::ics02_client::client_state::{AnyClientState, ClientState};
+use ibc::ics03_connection::connection::ConnectionEnd;
+use ibc::ics04_channel::error::Kind;
+use ibc::ics24_host::identifier::{ChainId, ChannelId, ClientId, ConnectionId, PortId};
+use ibc::Height;
+
+type Identifier = String;
+
+/// A cache storing filtering status (allow or deny) for
+/// arbitrary identifiers.
+#[derive(Default, Debug)]
+pub(crate) struct FilterCache {
+    /// A cache associating a generic identifier, such as
+    /// client id, channel id, or connection id, with a
+    /// boolean, representing either allow (if `true`) or
+    /// deny (if `false`) that identifier.
+    permission_cache: HashMap<Identifier, bool>,
```

Comment:
And instead of using a `String`, let's perhaps define an enum for the keys?

```rust
enum CacheKey {
  Client(ChainId, ClientId),
  Channel(ChainId, ChannelId),
  Connection(ChainId, ConnectionId),
}
```